src/colorizer/Dataset.ts

```diff
@@ -406,15 +405,42 @@ export default class Dataset {
     ]);
     const [outliers, tracks, times, centroids, bounds, _loadedFrame, ...featureResults] = result;
 
-    this.outliers = outliers;
-    this.trackIds = tracks;
-    this.times = times;
-    this.centroids = centroids;
-    this.bounds = bounds;
+    // TODO: Add reporting pathway for Dataset.load?
+    if (outliers.status === "rejected") {
+      console.warn("Failed to load outliers: ", outliers.reason);
+    } else {
+      this.outliers = outliers.value;
+    }
+    if (tracks.status === "rejected") {
+      console.warn("Failed to load tracks: ", tracks.reason);
+    } else {
+      this.trackIds = tracks.value;
+    }
+    if (times.status === "rejected") {
+      throw new Error("Time data could not be loaded. Is the dataset manifest file valid?");
+    } else {
+      this.times = times.value;
+    }
+    if (centroids.status === "rejected") {
+      console.warn("Failed to load centroids: ", centroids.reason);
+    } else {
+      this.centroids = centroids.value;
+    }
+    if (bounds.status === "rejected") {
+      console.warn("Failed to load bounds: ", bounds.reason);
+    } else {
+      this.bounds = bounds.value;
+    }
```

Comment:
```suggestion
    const successData = [this.outliners, this.trackIds, this.times, this centroids, this.bounds]
    results.forEach((result, index) => {
        if (result.status) == "rejected" {
            console.warn(`failed to load: ${result.reason}`
        } else if (index < successResults.length) {
            successResult[index] = result.value
         } else {
             const [key, data] = result.value;
             this.features.set(key, data)
         }
    
    }
```

Reply:
I wouldn't commit this, it was just easier to write out as code. It just seems like if you have an array as your return it's nice to continue to treat it as an array and reduce repeated code. If it's really important that you also have the name of the data in the warning message you could also make an array of names somewhere.

Reply:
Gotcha. Some of these types are required, so I'll need to add an extra step for whether we should throw an error if that data is missing. This makes sense though.

Reply:
Uhh, looking at this further I'm not sure this would actually work. This wouldn't play nicely with typing (different files are different types, including unsigned 8/16/32-bit integer arrays) and I don't think there's actually a way to keep references to data fields as an array and assign them like you show here unless I'm storing, say, a setter method.

I do get wanting to remove the repeated code here, so I tried adding a handler that will retrieve the promise value and throws a warning if it doesn't exist.